massa-protocol-worker/src/protocol_network.rs

```diff
@@ -117,8 +117,19 @@ impl ProtocolWorker {
             }
             NetworkEvent::ReceivedOperations { node, operations } => {
                 massa_trace!(OPS, { "node": node, "operations": operations});
-                self.on_operations_received(node, operations, op_timer)
-                    .await;
+                if operations.iter().any(|operation| {
```

Comment:
I think this should be better to be in `note_operations_from_node` here : https://github.com/massalabs/massa/blob/main/massa-protocol-worker/src/protocol_worker.rs#L932

Because is the function that is triggered each time we receive operation even if it's not in the propagation process. This will be more flexible and will be checked all time.

You don't ban in `note_operations_from_node` but you return an error and `on_operations_received` will ban